src/Test.UnitTests.Sarif.Multitool.Library/MergeCommandUnitTests.cs

```diff
@@ -0,0 +1,281 @@
+                {
+                    run.Results.Count.Should().Be(sarifLog1.Runs[1].Results.Count);
+                }
+                if (run.Tool.Driver.Name == "Tool3")
+                {
+                    run.Results.Count.Should().Be(sarifLog2.Runs[0].Results.Count);
+                }
+            }
+        }
+
+        [Fact]
+        public void MergeCommand_WhenMergeRunsOff_RunShouldAggregateByRuleToolVersion_SingleToolVersion()
+        {
+            // 2 logs 2 runs, same tool version, 6 rules
+            SarifLog sarifLog1 = new SarifLog { Runs = new[] { CreateTestRun(5) } };
+            SarifLog sarifLog2 = new SarifLog { Runs = new[] { CreateTestRun(6) } };
+            string sarifLog1Json = JsonConvert.SerializeObject(sarifLog1);
+            string sarifLog2Json = JsonConvert.SerializeObject(sarifLog2);
+            string sarifLog1FilePath = Path.Combine(testDirectory, "SarifLog1.sarif");
+            string sarifLog2FilePath = Path.Combine(testDirectory, "SarifLog2.sarif");
+            string outputFilePath = Path.Combine(testDirectory, "merged.sarif");
+            StringBuilder outputStringBuilder = new StringBuilder();
+
+            var mockFileSystem = new Mock<IFileSystem>();
+            ArrangeMockFileSystemRead(mockFileSystem, sarifLog1Json, sarifLog1FilePath);
+            ArrangeMockFileSystemRead(mockFileSystem, sarifLog2Json, sarifLog2FilePath);
+            ArrangeMockFileSystemCreate(mockFileSystem, outputFilePath, outputStringBuilder);
+            ArrangeMockFileSystemEnumerate(mockFileSystem, testDirectory, new[] { sarifLog1FilePath, sarifLog2FilePath });
+
+            IFileSystem fileSystem = mockFileSystem.Object;
+
+            var options = new MergeOptions
+            {
+                PrettyPrint = true,
+                OutputDirectoryPath = testDirectory,
+                OutputFileName = "merged.sarif",
+                TargetFileSpecifiers = new[] { "*.sarif" },
+                MergeRuns = false,
+                Force = true,
+            };
+
+            var mergeCommand = new MergeCommand(fileSystem);
+            int returnCode = mergeCommand.Run(options);
+            returnCode.Should().Be(0);
+
+            // merged log should have 6 runs grouped by rule id
+            SarifLog mergedLog = JsonConvert.DeserializeObject<SarifLog>(outputStringBuilder.ToString());
+            mergedLog.Runs.Count.Should().Be(6);
+            for (int i = 1; i <= 6; i++)
+            {
+                mergedLog.Runs[i - 1].Tool.Driver.Name.Should().Be("TestTool");
+                mergedLog.Runs[i - 1].Results[0].RuleId.Should().Be($"TESTRULE00{i}");
+            }
+        }
+
+        [Fact]
+        public void MergeCommand_WhenSplitPerRule_LogShouldAggregateByRuleToolVersion()
+        {
+            // 2 logs 2 runs, same tool version, 6 rules
+            SarifLog sarifLog1 = new SarifLog { Runs = new[] { CreateTestRun(6, null, true) } };
```

Comment:
>SarifLog [](http://example.com/codeflow?start=12&length=8)

this code doesn't meet our style standard, you should be using 'var' instead of an explicit type, because the type itself is clearly specified after the `new` keyword.

Why did this code pass build? Are we not enforcing this standard in our diagnostics configuration?

Reply:
Fixed. Its no enforced. In VS there are messages (not warning or erro) in error list showing this kind of issue.